lua/lspconfig/jsonnet_ls.lua

```diff
@@ -10,18 +10,20 @@ local function jsonnet_path(root_dir)
   return table.concat(paths, ':')
 end
 
+local function get_root_dir(fname)
+  return util.root_pattern('jsonnetfile.json')(fname) or util.find_git_ancestor(fname)
+end
+
 configs.jsonnet_ls = {
   default_config = {
     cmd = { 'jsonnet-language-server' },
     cmd_env = { JSONNET_PATH = vim.fn.getcwd() }, -- will be overwritten by `on_new_config`
     filetypes = { 'jsonnet', 'libsonnet' },
-    root_dir = vim.fn.getcwd(), -- will be overwritten by `on_new_config`
+    root_dir = get_root_dir,
     on_new_config = function(new_config, file_dir)
-      local root_dir = util.root_pattern('jsonnetfile.json')(file_dir) or util.find_git_ancestor(file_dir)
       new_config.cmd_env = {
-        JSONNET_PATH = jsonnet_path(root_dir)
+        JSONNET_PATH = jsonnet_path(get_root_dir(file_dir))
```

Comment:
Why do you have to do this again? file_dir is root_dir (I would rename this)

Reply:
What I get as root_dir is the directory of the opened file :shrug:

Reply:
then the root pattern is failing to find a root and you should get the same result again

Reply:
Or there is a bug, this is where it's supposed to be passed 

https://github.com/mjlbach/nvim-lspconfig/blob/c1af9bd8296b1627ed0517b1915c76545ea7cee9/lua/lspconfig/configs.lua#L208-L209

Reply:
You're right. I had an erroneous override of the default config in the setup function :man_facepalming: